mrp_timesheet_link/models/mrp_production.py

```diff
@@ -0,0 +1,43 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, fields, api, _
+
+
+class MRPProduction(models.Model):
+    _inherit = 'mrp.production'
+
+    project_id = fields.Many2one(comodel_name='project.project')
+    task_id = fields.Many2one(comodel_name='project.task')
+
+    timesheet_count = fields.Integer(compute='_compute_timesheet_count')
+
+    @api.onchange('project_id')
+    def _onchange_project_id(self):
+        self.task_id = False
+
+    def _compute_timesheet_count(self):
+        for record in self:
+            record.timesheet_count = len(record.task_id.timesheet_ids)
+
+    def write(self, vals):
+        if vals.get('task_id'):
```

Comment:
When production is cancelled task value is unset. Have you tested this code in that situation? Code suggest that this if should return `False` and shouldn't unset production link.

And this code should be iterable, since `write` is `multi`, direct `self` accesses could raise errors.